### PR TITLE
BAN-131: Remove native button implementation inside zen-button

### PR DESCRIPTION
**Pull request by f-jauregui**
[Jira ticket](https://reciprocitylabs.atlassian.net/browse/BAN-131)

### CODE REVIEW

File: src/components/zen-button/zen-button.tsx [hunk 1]
```diff
@@ -28,22 +28,29 @@ export class ZenButton {
...
-      <Host class={{ disabled: this.disabled }}>
-        <button type="button" class={{ btn: true, [`btn-${this.variant}`]: true }} disabled={this.disabled}>
-          <slot name="leadingIcon" />
-          {this.loading ? <zen-spinner></zen-spinner> : null}
-          {this.label ? (
-            <span class={{ ml: this.leadingIconSlotFulfilled, mr: this.trailingIconSlotFulfilled }}>{this.label}</span>
-          ) : null}
-          <slot name="trailingIcon" />
-        </button>
+      <Host class={{ btn: true, [`btn-${this.variant}`]: true, disabled: this.disabled }} tabindex={this.tabindex}>
```
Comment (chavlji) [inline src/components/zen-button/zen-button.tsx:47]: Will user be able to override this tabindex?

Comment (chavlji) [inline src/components/zen-button/zen-button.tsx:47]: Will user be able to override this tabindex if he'd set it as an attribute or should be present a prop?

**Review by chavlji — CHANGES_REQUESTED** [review]
Great, but needs some changes:
Tab to focus button. Press enter. onclick action should be triggered but doesn't.
Please add onclick event handler to primary story that shows some alert('clicked'). Will be easier to test.

File: src/components/zen-button/zen-button.tsx (hunk 1 above)
Comment (f-jauregui) [reply src/components/zen-button/zen-button.tsx:47]: Yes, it will.

**Review by chavlji — APPROVED** [review]
🚢 it!

**Pull request merged**